code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -1,202 +1,59 @@
-    <AppBadge
-      badgeContent={badgeContent}
-      variant="contained"
-      size="small"
-      className="header__toolbar-cart-badge"
-    >
-      <ShoppingCartIcon className="header__toolbar-icon" fontSize="medium" />
-    </AppBadge>
-  ) : (
-    <ShoppingCartIcon className="header__toolbar-icon" fontSize="medium" />
-  );
-
-  const loadingButton = isLoadingAuth && <AppLoader />;
-
-  const logoutButton = isAuthenticated && (
-    <AppButton
-      onClick={handleLogout}
-      variant="danger"
-      size="small"
-      data-testid="LogoutButton"
-    >
-      <LogoutIcon />
-    </AppButton>
-  );
-
-  const signInButton = (
-    <AppButton onClick={handleOpenAuthModal} data-cy="auth-button">
-      <AppTypography component="span" translationKey="signIn.label" />
-    </AppButton>
-  );
-
-  const loadingDashboardButton = isLoadingAuth && <AppLoader />;
-
-  const isDashboardAvailable =
-    (userRole === ROLES.SHOP_MANAGER || userRole === ROLES.ADMIN) &&
-    !isLoadingAuth;
-
-  const authenticatedDashboardButton = isDashboardAvailable && (
-    <AppTooltip titleTranslationKey="dashboard.tooltip">
-      <AppIconButton
-        to={routes.dashboard.path}
-        component={AppLink}
-        data-cy="dashboard-button"
-      >
-        <DashboardCustomizeIcon fontSize="medium" />
-      </AppIconButton>
-    </AppTooltip>
-  );
-
-  const dashboardButton =
-    loadingDashboardButton || authenticatedDashboardButton;
-
-  const authButton = loadingButton || logoutButton || signInButton;
-
-  const loadingOrdersButton = isLoadingAuth && <AppLoader />;
-
-  const authenticatedOrdersButton = userRole === ROLES.USER &&
-    !isLoadingAuth && (
-      <AppTooltip titleTranslationKey="orders.tooltip">
-        <AppIconButton
-          to={routes.orders.path}
-          component={AppLink}
-          data-cy="orders-button"
-        >
-          <ListAltIcon className="header__toolbar-icon" fontSize="medium" />
-        </AppIconButton>
-      </AppTooltip>
-    );
-
-  const ordersButton = loadingOrdersButton || authenticatedOrdersButton;
-
-  let cartButton: ReactNode = null;
-
-  if (isLoadingAuth) {
-    cartButton = <AppLoader />;
-  } else if (userRole === ROLES.USER || !isAuthenticated) {
-    cartButton = (
-      <AppTooltip titleTranslationKey="cart.tooltip">
-        <AppIconButton data-cy="header-cart-button" onClick={handleOpenCart}>
-          {badge}
-        </AppIconButton>
-      </AppTooltip>
-    );
+  const role = useUserRoleSelector();
+  const isAuthLoading = useIsAuthLoadingSelector();
+
+  let headerNavigation;
+
+  switch (role) {
+    case ROLES.ADMIN:
+      headerNavigation = <HeaderAdminToolbar />;
+      break;
+    case ROLES.SHOP_MANAGER:
+      headerNavigation = <HeaderShopManagerToolbar />;
+      break;
+    case ROLES.USER:
+      headerNavigation = <HeaderUserToolbar />;
+      break;
+    default:
+      headerNavigation = <HeaderUnauthorizedUserToolbar />;
```

Comment:
I suggest moving this function outside a component or even to a local util as I did in `DashboardLayout.tsx`. This util is local to the folder. You can create something similar to this:

```tsx
import DashboardAdminLayout from "@/layouts/dashboard-layout/components/dashboard-admin-tab-layout/DashboardAdminLayout";
import DashboardManagerLayout from "@/layouts/dashboard-layout/components/dashboard-manager-tab-layout/DashboardManagerLayout";

import { ROLES } from "@/constants/common";
import { UserRole } from "@/types/user.types";

const getDasboardLayoutByRole = (role: UserRole | null) => {
  switch (role) {
    case ROLES.ADMIN:
      return <DashboardAdminLayout />;
    case ROLES.SHOP_MANAGER:
      return <DashboardManagerLayout />;
    default:
      return false;
  }
};

export default getDasboardLayoutByRole;
```

Than we can use it similarly to following:

```tsx
const DashboardLayout = () => {
  const userRole = useUserRoleSelector();

  return getLayoutByRole(userRole);
};
```

Reply:
Thanks, done